### PR TITLE
Add pipeline for otel-collector image

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Changes proposed in this pull request:

- Create a new pipeline for the 3rd party otel-collector-image

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/15830

### CODE REVIEW

File: vpath/pjtester.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+pjConfigs:
+  prConfig: 
+    kyma-project:
+      third-party-images:
+        prNumber: 281
+  prowJobs:
+    kyma-project:
+      third-party-images:
+        - pjName: "pre-main-tpi-otel-collector"
+          report: true
```
Comment (Halamix2) [inline vpath/pjtester.yaml:10]: prConfig inside pjConfigs is only used to change base prowjob definition PR
```suggestion
pjConfigs:
  prowJobs:
    kyma-project:
      third-party-images:
        - pjName: "pre-main-tpi-otel-collector"
          report: true
 prConfigs:
    kyma-project:
      third-party-images:
        prNumber: 281
```

**Review by Halamix2 — CHANGES_REQUESTED** [review]

File: templates/data/third-party-images-data.yaml [hunk 2]
```diff
@@ -365,3 +365,32 @@ templates:
...
+              - jobConfig:
+                  name: "post-main-tpi-otel-collector"
+                  run_if_changed: "^otel-collector/"
+                  args:
+                    - "--name=tpi/otel-collector"
+                    - "--config=/config/kaniko-build-config.yaml"
+                    - "--context=otel-collector"
+                    - "--dockerfile=Dockerfile"
+                    - "--env-file=envs"
+                    - '--tag={{`{{ .Env \"KSM_VERSION\" }}-{{ .ShortSHA }}`}}'
```
Comment (Halamix2) [inline templates/data/third-party-images-data.yaml:391]: You probably want your image tag in `OTEL_VERSION`-`SHA` format, as you provide `OTEL_VERSION` through envs file in thire-party-images repo
```suggestion
                    - '--tag={{`{{ .Env \"OTEL_VERSION\" }}-{{ .ShortSHA }}`}}'
```

**Review by Halamix2 — CHANGES_REQUESTED** [review]

File: templates/data/third-party-images-data.yaml (hunk 2 above)
Comment (rakesh-garimella) [reply templates/data/third-party-images-data.yaml]: thanks fixed

**Review by Halamix2 — APPROVED** [review]

**Pull request merged**